server/application/application.go

```diff
@@ -1081,6 +1081,23 @@ func (s *Server) PodLogs(q *application.ApplicationPodLogsQuery, ws application.
 	if q.TailLines > 0 {
 		tailLines = &q.TailLines
 	}
+
+	type filterData struct {
```

Comment:
nitpick. I think struct is overkill here. The same can be implemented without it:

```golang
	literal := ""
	inverse := false
	if q.Filter != nil {
		literal = *q.Filter
		if literal[0] == '!' {
			literal = literal[1:]
			inverse = true
		}
	}
```